### PR TITLE
[16.0] [MIG] account_payment_batch_process

### PR DESCRIPTION
**Pull request by Murtaza-OSI**

### CODE REVIEW

**Comment by Gelo-fl** [comment]
@Murtaza-OSI Would it be possible to reopen the PR for validation?
Thanks in advance,

**Comment by Murtaza-OSI** [comment]
> @Murtaza-OSI Would it be possible to reopen the PR for validation? Thanks in advance,

I don't have access to opening PR

**Comment by ghost** [comment]
Hello @leemannd @dreispt, would it be possible to reopen the PR for a validation?

Thank you!

**Comment by leemannd** [comment]
Hello @ACheung-FactorLibre I'm PSC on oca/project I can't do anything here.

If you need it. You can take back the branch/commits and open a new pull-request with it.

**Review by dreispt — APPROVED** [review]

**Comment by ghost** [comment]
Thank you! @leemannd @dreispt

**Review by PauBForgeFlow — APPROVED** [review]
Functional review ok!

There is a display error in account.payment.register wizard.



Could you please check it?

**Comment by metaminux** [comment]
@StefanRijnhart or @dreispt Can one of you or someone with write access on OCA/account-payment merge this approved PR or clarify what to do with it and the concurrent one : #690 ?

Thanks a lot !

File: account_payment_batch_process/wizard/account_payment_register.py [hunk 1]
```diff
@@ -0,0 +1,563 @@
...
+            payment = (
+                self.env["account.payment"]  # pylint: disable=context-overridden
+                .with_context(context)
+                .create(self.get_payment_values(group_data=group_data[partner]))
+            )
+            payment_ids.append(payment.id)
+            payment.action_post()
+            # Reconciliation
+            domain = [
+                ("account_type", "in", ("receivable", "payable")),
```
Comment (StefanRijnhart) [inline account_payment_batch_process/wizard/account_payment_register.py:421]: Looks like `account.payment`'s `line_ids` in Odoo 16 refer to the `account.move.line`s of the inherited `account.move`. `account_type` on `account.move.line` is a related field on `account.account`'s `account_type`. These don't seem to be valid values for `account_type`. The values `receivable` and `payable` should be updated to reflect the valid values in this selection: https://github.com/odoo/odoo/blob/16.0/addons/account/models/account_account.py#L44-L64

**Comment by StefanRijnhart** [comment]
I'm rather troubled by the fact that there isn't a single test in this module. Can you add one?

File: account_payment_batch_process/wizard/account_payment_register.py (hunk 1 above)
Comment (StefanRijnhart) [reply account_payment_batch_process/wizard/account_payment_register.py:421]: https://github.com/OCA/account-payment/pull/690 does not have this issue BTW. Maybe go with that one?

**Comment by etobella** [comment]
Also, there is a comment to fix from @PauBForgeFlow

**Comment by StefanRijnhart** [comment]
I imagine that one is also fixed by https://github.com/OCA/account-payment/pull/690/commits/7a763b95a9e2a2b57e586f97a826c909a2401f0c#diff-530410616a3c4deadb98758e38062cdfa4ccc60b77fd306324970091300def78L57-R68
Closing this one in favour of #690.

**Pull request closed without merging**